examples/restaurant_demo/restaurant.py

```diff
@@ -0,0 +1,341 @@
+        
+        Before performing the computation, 
+        we pre-aggregate each visitor' spending for the day. Return a dictionary mapping days to revenue
+        """
+
+        # Pre-process the data set: limit the number of days contributed by a visitor to SUM_MAX_CONTRIBUTED_DAYS
+        day_visits = bound_visits_per_week(self._day_visits, SUM_MAX_CONTRIBUTED_DAYS)
+
+        day_revenue = dict()
+
+        # Use the default epsilon value if it is not given as an argument
+        if not epsilon:
+            x = BoundedSum(
+                self._epsilon,
+                MIN_EUROS_SPENT,
+                MAX_EUROS_SPENT_2,
+                l0_sensitivity=SUM_MAX_CONTRIBUTED_DAYS,
+            )
+        else:
+            x = BoundedSum(
+                epsilon,
+                MIN_EUROS_SPENT,
+                MAX_EUROS_SPENT_2,
+                l0_sensitivity=SUM_MAX_CONTRIBUTED_DAYS,
+            )
+
+        for day in day_visits["Day"].unique():
+            # For each visitor, pre-aggregate their spending for the day.
+            visits_on_day = day_visits[day_visits["Day"] == day][
+                ["VisitorId", "Money spent (euros)"]
+            ]
+            visitor_to_spending = dict()
+
+            for visitor in visits_on_day["VisitorId"].unique():
+                visitor_to_spending[visitor] = visits_on_day[
+                    visits_on_day["VisitorId"] == visitor
+                ]["Money spent (euros)"].sum()
+
+            spending = list(visitor_to_spending.values())
+
+            x.reset()
+            # Can use either quick_result or a combination of add_entries() and result()
+            x.add_entries(data=spending)
+            day_revenue[day] = int(x.result())
+
+        return day_revenue
+
+
+def bound_visits_per_week(df, limit):
+    """Utility function to modify the input dataframes. 
+    
+    Makes sure that each visitor only contributes at most 3 visits per week,
+    Returns an updated dataframe 
+    """
+
+    # Shuffle dataframe
+    df = df.sample(frac=1)
+
+    # Store results in new dataframe
+    updated_df = pd.DataFrame(columns=df.columns)
+    df_idx = 0
+
+    # Keep track of visits per visitor and make sure it doesn't go above the limit
+    # {visitor id: # of times it has visited the restaurant}
+    id_to_visited = defaultdict(int)
+
+    for index in df.index:
+        if id_to_visited[df.loc[index]["VisitorId"]] < limit:
+            updated_df.loc[df_idx] = df.loc[index]
+            df_idx += 1
+            id_to_visited[df.loc[index]["VisitorId"]] += 1
+
+    return updated_df
+
+
+if __name__ == "__main__":
+    r = RestaurantStatistics(
+        hours_filename="day_data.csv", days_filename="week_data.csv"
+    )
+
+    np_count_hour, p_count_hour = r.count_visits_per_hour()
+    print("-----------------------------------")
+    print("Visits per hour:")
+    print("Non-private:", np_count_hour)
```

Comment:
Suggestion:

Maybe instead of saying "Non-private:" and "Private:" in print statements, we can say "Without differential privacy:" and "With differential privacy:", respectively? ("Private" can have many meanings.)

Reply:
You are absolutely right, done!